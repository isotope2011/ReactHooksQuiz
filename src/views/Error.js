import React from "react";

export default ({ errors }) => {
  return (
    <>
      {errors ? (
        <div style={{ color: "red" }}>{errors}</div>
      ) : (
        <div>Something went wrong...</div>
      )}
    </>
  );
};
