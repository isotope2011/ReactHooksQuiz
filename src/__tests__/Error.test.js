import React from 'react';
import { render, screen } from '@testing-library/react';
import snapshotDiff from 'snapshot-diff';

import Error from '../views/Error';

describe('Error View Component', () => {
    test('Display Error Message', () => {
        const { getByText } = render(<Error {...{ errors: 'Error Message' }} />);
        expect(getByText('Error Message')).toMatchSnapshot();
    })

    test('Test using asFragment for first render check', () => {
        const { asFragment } = render(<Error {...{ errors: 'Error Message' }} />);
        const firstRender = asFragment();
        // expect(firstRender).toMatchDiffSnapshot(asFragment());
        // using snapshotDiff
        expect(
            snapshotDiff(firstRender, asFragment())
        ).toMatchSnapshot();
    })

    test('Show Something went wrong... as default', () => {
        render(<Error />);
        expect(screen.getByText('Something went wrong...')).toMatchSnapshot();
    })
});

