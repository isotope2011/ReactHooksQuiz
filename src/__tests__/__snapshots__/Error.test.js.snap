// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error View Component Display Error Message 1`] = `
<div
  style="color: red;"
>
  Error Message
</div>
`;

exports[`Error View Component Show Something went wrong... as default 1`] = `
<div>
  Something went wrong...
</div>
`;

exports[`Error View Component Test using asFragment for first render check 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
